utils/shared.js

```diff
@@ -757,6 +757,18 @@ const isDateTimeFormat = (value) => {
     return typeof value == "string" && (/\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{3}Z/.test(value));
 }
 
+const redactEmailLoginInfo = (loginDetails) => {
+    let amp = loginDetails.indexOf('@');    
+    for(let i = 0; i < amp; i++) {
+        if(i != 0 && i != 1 && i != amp - 1) {
+            loginDetails = loginDetails.substring(0, i) + "*" + loginDetails.substring(i + 1);
+        } 
+    }
+    return loginDetails
+}
+
+const redactPhoneLoginInfo = (loginDetails) => { return loginDetails = "***-***-" + loginDetails.substring(loginDetails.length - 4); }
```

Comment:
I used `loginDetails = "***-***-" + fetchedData[348474836].slice(-4);`

Reply:
This is something simpler tho.